src/ORM/Behavior/CounterCacheBehavior.php

```diff
@@ -219,11 +219,24 @@ protected function _processAssociation(
         array $settings
     ): void {
         $foreignKeys = (array)$assoc->getForeignKey();
-        $primaryKeys = (array)$assoc->getBindingKey();
         $countConditions = $entity->extract($foreignKeys);
+        $allNulls = true;
+        foreach ($countConditions as $field => $value) {
```

Comment:
This loop is not pretty but it's to account for multi field foreign key with a mix of null and non null values.